haystack/document_stores/memory.py

```diff
@@ -146,6 +172,25 @@ def write_documents(
                     )
                     continue
             self.indexes[index][document.id] = document
+            modified_documents += 1
+
+        if self.use_bm25 is True and modified_documents > 0:
+            self.update_bm25(index=index)
+
+    def update_bm25(self, index: Optional[str] = None):
+        """
+        Updates the BM25 sparse representation in the the document store.
+
+        :param index: Index name for which the BM25 representation is to be updated. If set to None, the default self.index is used.
+        """
+        index = index or self.index
+        tokenizer = re.compile(self.bm25_tokenization_regex).findall
+
+        logger.info("Updating BM25 representation...")
+        all_documents = self.get_all_documents(index=index)
+        tokenized_corpus = [tokenizer(doc.content.lower()) for doc in all_documents]
+        bm25_class = getattr(rank_bm25, self.bm25_algorithm)
+        self.bm25[index] = bm25_class(tokenized_corpus, **self.bm25_parameters)
```

Comment:
If we change line 106 as suggested, this would become
```suggestion
        self.bm25[index] = self.bm25_algorithm(tokenized_corpus, **self.bm25_parameters)
```